js/pro/woo.js

```diff
@@ -0,0 +1,722 @@
+                'tradesLimit': 1000,
+                'ordersLimit': 1000,
+                'requestId': {},
+            },
+            'streaming': {
+                'ping': this.ping,
+                'keepAlive': 10000,
+            },
+        });
+    }
+
+    requestId (url) {
+        const options = this.safeValue (this.options, 'requestId', {});
+        const previousValue = this.safeInteger (options, url, 0);
+        const newValue = this.sum (previousValue, 1);
+        this.options['requestId'][url] = newValue;
+        return newValue;
+    }
+
+    async watchPublic (messageHash, message) {
+        this.checkRequiredUid ();
+        const url = this.urls['api']['ws']['public'] + '/' + this.uid;
+        const requestId = this.requestId (url);
+        const subscribe = {
+            'id': requestId,
+        };
+        const request = this.extend (subscribe, message);
+        return await this.watch (url, messageHash, request, messageHash, subscribe);
+    }
+
+    async watchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const name = 'orderbook';
+        const market = this.market (symbol);
+        const topic = market['id'] + '@' + name;
+        const request = {
+            'event': 'subscribe',
+            'topic': topic,
+        };
+        const message = this.extend (request, params);
+        const orderbook = await this.watchPublic (topic, message);
+        return orderbook.limit (limit);
+    }
+
+    handleOrderBook (client, message) {
+        //
+        //     {
+        //         topic: 'PERP_BTC_USDT@orderbook',
+        //         ts: 1650121915308,
+        //         data: {
+        //             symbol: 'PERP_BTC_USDT',
+        //             bids: [
+        //                 [
+        //                     0.30891,
+        //                     2469.98
+        //                 ]
+        //             ],
+        //             asks: [
+        //                 [
+        //                     0.31075,
+        //                     2379.63
+        //                 ]
+        //             ]
+        //         }
+        //     }
+        //
+        const data = this.safeValue (message, 'data');
+        const marketId = this.safeString (data, 'symbol');
+        const market = this.safeMarket (marketId);
+        const symbol = market['symbol'];
+        const topic = this.safeString (message, 'topic');
+        let orderbook = this.safeValue (this.orderbooks, symbol);
+        if (orderbook === undefined) {
+            orderbook = this.orderBook ({});
+        }
+        const timestamp = this.safeInteger (message, 'ts');
+        const snapshot = this.parseOrderBook (data, symbol, timestamp, 'bids', 'asks');
+        orderbook.reset (snapshot);
+        client.resolve (orderbook, topic);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const name = 'ticker';
```

Comment:
@sc0Vu mainnet seems down,  is just me?
![image](https://user-images.githubusercontent.com/43336371/205094578-f37ba2d3-cbdb-4bb3-8632-9a660d69923e.png)

Reply:
Not sure about this, but test net works for me.